frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1258,15 +1265,23 @@ export const surveyLogic = kea<surveyLogicType>([
 
                     const data: number[] = questionResults.data
                     if (data.length === 11) {
-                        const promoters = data.slice(9, 11).reduce((a, b) => a + b, 0)
-                        const passives = data.slice(7, 9).reduce((a, b) => a + b, 0)
-                        const detractors = data.slice(0, 7).reduce((a, b) => a + b, 0)
-                        const npsScore = ((promoters - detractors) / (promoters + passives + detractors)) * 100
-                        return npsScore.toFixed(1)
+                        return calculateNpsScore(calculateNpsBreakdown(questionResults)).toFixed(1)
                     }
                 }
             },
         ],
+        npsBreakdown: [
+            (s) => [s.surveyRatingResults],
+            (surveyRatingResults) => {
+                if (!surveyRatingResults) {
```

Comment:
we should check `Object.keys(surveyRatingResults)` length as well, if its 0, its not gonna work

Reply:
agree, will update